diff --git a/modules/libpref/Preferences.cpp b/modules/libpref/Preferences.cpp
index 82c9afe448..813e690858 100644
--- a/modules/libpref/Preferences.cpp
+++ b/modules/libpref/Preferences.cpp
@@ -3580,24 +3580,9 @@ static const char kChannelPref[] = "app.update.channel";
 #ifdef MOZ_WIDGET_ANDROID

 static Maybe<bool> TelemetryPrefValue() {
-  // Leave it unchanged if it's already set.
-  // XXX: how could it already be set?
-  if (Preferences::GetType(kTelemetryPref) != nsIPrefBranch::PREF_INVALID) {
-    return Nothing();
-  }
-
-  // Determine the correct default for toolkit.telemetry.enabled. If this
-  // build has MOZ_TELEMETRY_ON_BY_DEFAULT *or* we're on the beta channel,
-  // telemetry is on by default, otherwise not. This is necessary so that
-  // beta users who are testing final release builds don't flipflop defaults.
-#  ifdef MOZ_TELEMETRY_ON_BY_DEFAULT
-  return Some(true);
-#  else
-  nsAutoCString channelPrefValue;
-  Unused << Preferences::GetCString(kChannelPref, channelPrefValue,
-                                    PrefValueKind::Default);
-  return Some(channelPrefValue.EqualsLiteral("beta"));
-#  endif
+  // LACUNA PATCH: Force telemetry default to OFF at compile time
+  return Some(false);
+  // END LACUNA PATCH
 }

 /* static */
@@ -3614,36 +3599,8 @@ void Preferences::SetupTelemetryPref() {
 #else  // !MOZ_WIDGET_ANDROID

 static bool TelemetryPrefValue() {
-  // For platforms with Unified Telemetry (here meaning not-Android),
-  // toolkit.telemetry.enabled determines whether we send "extended" data.
-  // We only want extended data from pre-release channels due to size.
-
-  constexpr auto channel = MOZ_STRINGIFY(MOZ_UPDATE_CHANNEL) ""_ns;
-
-  // Easy cases: Nightly, Aurora, Beta.
-  if (channel.EqualsLiteral("nightly") || channel.EqualsLiteral("aurora") ||
-      channel.EqualsLiteral("beta")) {
-    return true;
-  }
-
-#  ifndef MOZILLA_OFFICIAL
-  // Local developer builds: non-official builds on the "default" channel.
-  if (channel.EqualsLiteral("default")) {
-    return true;
-  }
-#  endif
-
-  // Release Candidate builds: builds that think they are release builds, but
-  // are shipped to beta users.
-  if (channel.EqualsLiteral("release")) {
-    nsAutoCString channelPrefValue;
-    Unused << Preferences::GetCString(kChannelPref, channelPrefValue,
-                                      PrefValueKind::Default);
-    if (channelPrefValue.EqualsLiteral("beta")) {
-      return true;
-    }
-  }
-
+  // LACUNA PATCH: Force telemetry default to OFF at compile time
+  // END LACUNA PATCH
   return false;
 }

