diff --git a/browser/components/urlbar/UrlbarUtils.sys.mjs b/browser/components/urlbar/UrlbarUtils.sys.mjs
index b7beb23cdd..2d28b861b1 100644
--- a/browser/components/urlbar/UrlbarUtils.sys.mjs
+++ b/browser/components/urlbar/UrlbarUtils.sys.mjs
@@ -1157,12 +1157,15 @@ export var UrlbarUtils = {
   addToFormHistory(input, value, source) {
     // If the user types a search engine alias without a search string,
     // we have an empty search string and we can't bump it.
-    // We also don't want to add history in private browsing mode.
+    // We also don't want to add history in private browsing mode or when
+    // browser.formfill is disabled.
     // Finally we don't want to store extremely long strings that would not be
     // particularly useful to the user.
+    // FIREBOX PATCH: Respect browser.formfill.enable setting
     if (
       !value ||
       input.isPrivate ||
+      !lazy.FormHistory.enabled ||
       value.length >
         lazy.SearchSuggestionController.SEARCH_HISTORY_MAX_VALUE_LENGTH
     ) {
